### PR TITLE
Improve BadRequestError error string

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/errors.py [hunk 1]
```diff
@@ -308,7 +308,27 @@ def __str__(self) -> str:
 
         value = super().__str__()
         if self.errors:
-            value += "\n" + data_binding.dump_json(self.errors, indent=2)
+            value += "\n\n"
+
+            def dump(obj: data_binding.JSONObject, obj_string: str = "") -> str:
```
Comment (FasterSpeeding) [inline hikari/errors.py:313]: Does this function have to be defined inline and can thus logic just be merged into the parent function?

File: hikari/errors.py [hunk 2]
```diff
@@ -308,7 +308,27 @@ def __str__(self) -> str:
...
+                string = ""
+                for k, v in obj.items():
+                    if isinstance(v, typing.Sequence):
+                        errors_string = obj_string + ":"
+
+                        for item in v:
+                            errors_string += f"\n - {item['message']}"
```
Comment (FasterSpeeding) [inline hikari/errors.py:320]: Might be worth it not assuming the error will always be in this format and instead falling back to a more basic handling

File: hikari/errors.py (hunk 1 above)
Comment (davfsa) [reply hikari/errors.py]: Extracted the logic

File: hikari/errors.py (hunk 2 above)
Comment (davfsa) [reply hikari/errors.py]: Added a fallback for when a KeyError occurs to just dump the errors JSON (the previous behaviour). Personally I think that the errors structure is well defined enough in the docs for this not be an issue (can also be seen in the wild), but better safe than sorry ig.

Also, TIL that `str` failing when raising an exception will just add `str failed` as the str of the exception

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**